docs/adr/0002-custom-dev-container-image.md

```diff
@@ -0,0 +1,34 @@
+# 2. Provide a custom dev container image
+
+Date: 2024-06-04
+
+## Status
+
+Accepted
+
+## Context
+
+We are using [dev containers](https://code.visualstudio.com/docs/devcontainers/create-dev-container) to provide a configuration for Codespaces that gives researchers a useful development environment to use while working on their OpenSAFELY projects. We want to improve that environment to make it more appealing and useful to researchers.
```

Comment:
I think we can rely on ADR 0001 (#43) to give the motivation for configuring a standard development environment. Here, I feel the context is that a dev container needs an image, but the choice of image is ours.

Ultimately, the choice of image rests on the features we want the standard development environment to have. Prior to the Codespaces initiative, the ehrQL tutorial and the getting started tutorial used a standard image (python:3.11-bullseye). So, a standard image had _some_ of the features we wanted. What changed with the Codespaces initiative? Why?

As I mentioned in #43, Nygard states that this section should describe "the forces at play". The tension between using a standard image and using a custom image is a force.

Reply:
I think I've now done a better job of explaining this in the context sections, but let me know if not.